Lesson_16/l16_h01_class.py

```diff
@@ -0,0 +1,77 @@
+"""
+HW:16[Volkov Anton].
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters
+# do not forget about block if __name__ == '__main__':
+"""
+
+
+class MonoBankCard:
+    """Class for creating cards."""
+
+    _lvl_card = None
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date):
+        """Initialize class."""
+        self._card_num = card_num
+        self._cvv = cvv
+        self._pin = pin
+        self._name = name
+        self._surname = surname
+        self._end_date = end_date
+        self._lvl_card = 'Universal'
+
+    @classmethod
+    def get_lvl_card(cls):
+        """Check the level of the card."""
+        return cls._lvl_card
+
+    @staticmethod
+    def max_lvl_card():
+        """Return the maximum level of the card."""
+        return 'Platinum'
+
+    @property
+    def card_num(self):
+        """Get the card number."""
+        return self._card_num
+
+    @card_num.setter
+    def card_num(self, new_card_num):
+        """Set the card number."""
+        self._card_num = new_card_num
+
+    @property
+    def pin(self):
```

Comment:
Not critical but in property you could add some verification , for example that pin has only digits and has digits not less that certain pin_min_size.